modules/ingest-common/src/main/java/org/opensearch/ingest/common/IngestCommonModulePlugin.java

```diff
@@ -147,4 +157,27 @@ private static MatcherWatchdog createGrokThreadWatchdog(Processor.Parameters par
         );
     }
 
+    private Map<String, Processor.Factory> filterForAllowlistSetting(Settings settings, Map<String, Processor.Factory> map) {
+        if (PROCESSORS_ALLOWLIST_SETTING.exists(settings) == false) {
+            return Map.copyOf(map);
+        }
+        final Set<String> allowlist = Set.copyOf(PROCESSORS_ALLOWLIST_SETTING.get(settings));
+        // Assert that no unknown processors are defined in the allowlist
+        final Set<String> unknownAllowlistProcessors = allowlist.stream()
+            .filter(p -> map.containsKey(p) == false)
+            .collect(Collectors.toSet());
```

Comment:
We can filter directly on map's key to find the unknown processor. This avoids iterating over allowlist
```suggestion
        Set<String> unknownAllowlistProcessors = map.keySet().stream()
        .filter(processor -> !allowlist.contains(processor))
        .collect(Collectors.toUnmodifiableSet());

```

Reply:
I want to find all keys in the allowlist that are not in the map. I think your suggested code is finding all keys in the map that are not in the allowlist.

Reply:
Ah! Yeah, I misinterpreted that. Do you want to change `toSet` to `toUnmodifiableSet` or leave it as it is?

Reply:
I'll change it in a follow up PR. Thanks!